src/mocks/aktivitet.js

```diff
@@ -514,55 +502,102 @@ export function opprettAktivitet(pathParams, body) {
         opprettetDato: new Date(),
         lagtInnAv: bruker,
         endretDato: moment().toISOString(),
-        endretAv: null,
+        endretAv: bruker,
         versjon: '1',
         erLestAvBruker: eksternBruker,
+        transaksjonsType: 'OPPRETTET',
         ...body,
     });
     aktiviteter.push(newAktivitet);
+    versjoner.push(newAktivitet);
     return newAktivitet;
 }
 
-export function oppdaterAktivitet({ aktivitetId }, aktivitet) {
-    const oldAktivitet = aktiviteter.find((aktivitet) => aktivitet.id === aktivitetId);
-    Object.assign(oldAktivitet, aktivitet);
-    oldAktivitet.endretDato = moment().toISOString();
-    oldAktivitet.endretAv = bruker;
-    oldAktivitet.lagtInnAv = bruker;
+function doOppdaterInternMockStateOgReturnerNyAktivitet(aktivitetId, nyeAktivitetAttributter) {
+    // Hent den gamle aktiviteten
+    const gammelAktivitet = aktiviteter.find((a) => a.id === aktivitetId);
+    // Merge de nye attributtene inn i den originale aktiviteten
+    const nyAktivitet = Object.assign({}, gammelAktivitet, lagNyVersion(nyeAktivitetAttributter));
+    // Legg til ny versjon i historikk
+    versjoner.push(nyAktivitet);
+    // Overskriv eksisterende aktivitet i aktiviteterData
+    Object.assign(gammelAktivitet, nyAktivitet);
+    return nyAktivitet;
+}
 
-    return oldAktivitet;
+function lagNyVersion(aktivitet) {
+    return {
+        ...aktivitet,
+        versjon: aktivitet.versjon + 1,
```

Comment:
Bare en heads up. Versjon er faktisk et løpenummer og er unikt og kunne vært brukt som primary key i aktivitet tabellen. Det virker logisk at aktivitetversjonen skal økes med 1 for hver gang men det kan vel så ofte økes med mer, hvis en annen aktivitet på en hvilken som helst bruker har vært oppdatert i mellomtiden. Skjønner godt at du ville gjøre det sånn men er usikker på om det vil gjøre at andre som ikke kjenner backenden får forsterket sin tro om at ting fungerer på en logisk måte når det egentlig ikke gjør det.

Reply:
Vi lar det være slik for nå, da det blir mye ekstraarbeid å introdusere en global sekvensgenerator for disse. Teknisk sett vil i hvert fall denne løsningen fungere for alle praktiske formål.